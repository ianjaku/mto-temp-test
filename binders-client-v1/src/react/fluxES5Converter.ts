/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
