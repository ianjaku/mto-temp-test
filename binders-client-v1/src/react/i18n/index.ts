import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
