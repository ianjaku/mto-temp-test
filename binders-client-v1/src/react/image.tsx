/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
