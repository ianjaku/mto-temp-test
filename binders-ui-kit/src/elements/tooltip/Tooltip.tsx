import * as React from "react";
import * as ReactDOM from "react-dom";
import { createPortal } from "react-dom";
import cx from "classnames";
import "./tooltip.styl";

export interface ITooltipProps {
    message: string;
    fadeIn?: number;
    fadeOnTimeout?: boolean;
    arrowOnBottom?: boolean;
    respectLineBreaks?: boolean;
}

export interface ITooltipState {
    top: string;
    left: string;
    show: boolean;
}

export class Tooltip extends React.Component<ITooltipProps, ITooltipState> {

    public static defaultProps = {
        fadeIn: 2000,
        fadeOnTimeout: true,
    };

    private timeoutId: ReturnType<typeof setTimeout>;
    private component: React.ReactInstance;
    private mounted: boolean;

    constructor(props: ITooltipProps) {
        super(props);

        this.showTooltipAt = this.showTooltipAt.bind(this);
        this.hideTooltip = this.hideTooltip.bind(this);

        this.state = {
            left: "0px",
            show: false,
            top: "0px",
        };
    }
    public componentDidMount(): void {
        this.mounted = true;
    }
    public componentWillUnmount(): void {
        this.mounted = false;
    }
    public componentDidUpdate(_prevProps: ITooltipProps, prevState: ITooltipState): void {
        const { fadeOnTimeout } = this.props;
        if (fadeOnTimeout && !prevState.show && this.state.show) {
            this.timeoutId = setTimeout(this.hideTooltip, this.props.fadeIn);
        }
    }

    public getWidth(): number {
        const node = ReactDOM.findDOMNode(this.component) as HTMLElement;
        return node.getBoundingClientRect().width;
    }

    private setStateIfMounted(stateUpdate: Partial<ITooltipState>): void {
        if (this.mounted) {
            this.setState(stateUpdate as ITooltipState);
        }
    }

    public showTooltipAt(top: string, left: string): void {
        const show = !this.state.show;
        clearTimeout(this.timeoutId);
        this.setStateIfMounted({ top, left, show });
    }

    public hideTooltip(): void {
        clearTimeout(this.timeoutId);
        this.setStateIfMounted({ show: false });
    }

    public render(): JSX.Element {
        const { message } = this.props;
        const { top, left, show } = this.state;
        const visibility = show ? "visible" : "hidden";

        return createPortal(
            <div
                ref={ref => { this.component = ref; }}
                className={cx("tooltip", { "tooltip--respectLineBreaks": this.props.respectLineBreaks })}
                style={{ top, left, visibility }}
            >
                {this.props.arrowOnBottom && (
                    <div className="tooltip-arrowOnBottom" />
                )}
                <span>{message}</span>
            </div>
            , document.body);
    }
}

export enum TooltipPosition {
    TOP,
    LEFT,
    RIGHT,
    BOTTOM,
}

export function showTooltip(
    e: React.MouseEvent<HTMLElement>,
    tooltip: Tooltip,
    position = TooltipPosition.BOTTOM,
    nudgePosition: { top?: number, left?: number } = { top: 0, left: 0 },
): void {
    const caller = e.target as HTMLElement;
    const viewportOffset = caller.getBoundingClientRect();
    showTooltipForBoundingClientRect(viewportOffset, tooltip, position, nudgePosition);
}

export function showTooltipForBoundingClientRect(
    rect: DOMRect,
    tooltip: Tooltip,
    position = TooltipPosition.BOTTOM,
    nudgePosition: { top?: number, left?: number } = { top: 0, left: 0 },
): void {
    const { top: calculatedTop, left: calculatedLeft } = calculateTooltipCoordinates(tooltip, rect, position);
    const top = calculatedTop + (nudgePosition.top ?? 0);
    const left = calculatedLeft + (nudgePosition.left ?? 0);
    if (tooltip && (top > 0 || left > 0)) {
        tooltip.showTooltipAt(`${top}px`, `${left}px`);
    }
}

export function hideTooltip(_e: React.MouseEvent<HTMLElement>, tooltip: Tooltip): void {
    if (tooltip) {
        tooltip.hideTooltip();
    }
}

function calculateTooltipCoordinates(tooltip: Tooltip, rect: DOMRect, position: TooltipPosition, it = 0): { top: number; left: number; } {
    if (it > 1 || !tooltip) {
        return { top: 0, left: 0 };
    }

    const { innerWidth, innerHeight } = window;
    const { top, left, height, width } = rect;
    const tooltipWidth = tooltip.getWidth();

    switch (position) {
        case TooltipPosition.TOP:
            return top - height > 0 ?
                { top: top - height - 10, left } :
                calculateTooltipCoordinates(tooltip, rect, TooltipPosition.BOTTOM, it + 1);
        case TooltipPosition.LEFT:
            return left - tooltipWidth > 0 ?
                { top, left: left - tooltipWidth } :
                calculateTooltipCoordinates(tooltip, rect, TooltipPosition.RIGHT, it + 1);
        case TooltipPosition.RIGHT:
            return left + width < innerWidth ?
                { top, left: left + width } :
                calculateTooltipCoordinates(tooltip, rect, TooltipPosition.LEFT, it + 1);
        default:
            return top + height < innerHeight ?
                { top: top + height, left } :
                calculateTooltipCoordinates(tooltip, rect, TooltipPosition.TOP, it + 1);
    }
}

export default Tooltip;
