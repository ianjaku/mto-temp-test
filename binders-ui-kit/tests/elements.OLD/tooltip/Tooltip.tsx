import * as React from "react";
import Tooltip, { TooltipPosition, showTooltip } from "../../../src/elements/tooltip/Tooltip";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const MESSAGE = "Hello, I'm a tooltip!";
const BOUNDING_RECT = { width: 100, height: 50, left: 110, top: 60, right: 0, bottom: 0 };

const createTooltip = () => wrapWithTheme(<Tooltip message={MESSAGE} />);
const createActionableTooltip = (position: TooltipPosition = TooltipPosition.TOP) => (
    wrapWithTheme(<ActionableTooltip position={position} />)
);


describe("Tooltip", () => {
    test("Tooltip (snapshot)", () => {
        const tooltip = create(createTooltip());
        const serialized = tooltip.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Actionable Tooltip (snapshot)", () => {
        const tooltip = create(createActionableTooltip());
        const serialized = tooltip.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Tooltip text (mount: enzyme+jsdom)", () => {
        const tooltip = mount(createTooltip());
        const tooltipElement = tooltip.find(".tooltip > span").at(0);
        const tooltipText = tooltipElement.text();
        expect(tooltipText).toEqual(MESSAGE);
    });

    test("Tooltip visibility toggle (mount: enzyme+jsdom)", () => {
        const component = mount(createActionableTooltip());
        const style = getMockedStyleForTooltip(component);

        expect(style).toHaveProperty("visibility", "visible");
    });

    test("Tooltip top position (mount: enzyme+jsdom)", () => {
        const component = mount(createActionableTooltip());
        const style = getMockedStyleForTooltip(component);
        const top = BOUNDING_RECT.top - BOUNDING_RECT.height;
        expect(style).toHaveProperty("top", `${top}px`);
        expect(style).toHaveProperty("left", `${BOUNDING_RECT.left}px`);
    });

    test("Tooltip left position (mount: enzyme+jsdom)", () => {
        const component = mount(createActionableTooltip(TooltipPosition.BOTTOM));
        const style = getMockedStyleForTooltip(component);
        const top = BOUNDING_RECT.top + BOUNDING_RECT.height;
        expect(style).toHaveProperty("top", `${top}px`);
        expect(style).toHaveProperty("left", `${BOUNDING_RECT.left}px`);
    });

    test("Tooltip left position (mount: enzyme+jsdom)", () => {
        const component = mount(createActionableTooltip(TooltipPosition.LEFT));
        const style = getMockedStyleForTooltip(component);

        expect(style).toHaveProperty("top", `${BOUNDING_RECT.top}px`);
        expect(style).toHaveProperty("left", `${BOUNDING_RECT.left}px`);
    });

    test("Tooltip left position (mount: enzyme+jsdom)", () => {
        const component = mount(createActionableTooltip(TooltipPosition.RIGHT));
        const style = getMockedStyleForTooltip(component);
        const left = BOUNDING_RECT.left + BOUNDING_RECT.width;
        expect(style).toHaveProperty("top", `${BOUNDING_RECT.top}px`);
        expect(style).toHaveProperty("left", `${left}px`);
    });
});

function getMockedStyleForTooltip(component): void {
    const button = component.find("button").get(0);
    button.props.onClick({ target: { getBoundingClientRect: () => BOUNDING_RECT } });
    component.update();
    return component.find(".tooltip").get(0).props.style;
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
class ActionableTooltip extends React.Component<{ position: TooltipPosition; }, any> {
    private tooltip: Tooltip;

    constructor(props) {
        super(props);

        this.onClick = this.onClick.bind(this);
    }

    public onClick(e) {
        const { position } = this.props;
        showTooltip(e, this.tooltip, position);
    }

    public render() {
        return (
            <div>
                <Tooltip ref={ref => { this.tooltip = ref; }} message={MESSAGE} />
                <button onClick={this.onClick}>
                    Show tooltip
                </button>
            </div>
        );
    }
}
