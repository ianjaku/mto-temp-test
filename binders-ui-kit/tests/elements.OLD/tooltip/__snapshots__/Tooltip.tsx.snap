// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Actionable Tooltip (snapshot) 1`] = `
<div>
  <div
    className="tooltip"
    style={
      Object {
        "left": "0px",
        "top": "0px",
        "visibility": "hidden",
      }
    }
  >
    <span>
      Hello, I'm a tooltip!
    </span>
  </div>
  <button
    onClick={[Function]}
  >
    Show tooltip
  </button>
</div>
`;

exports[`Tooltip Tooltip (snapshot) 1`] = `
<div
  className="tooltip"
  style={
    Object {
      "left": "0px",
      "top": "0px",
      "visibility": "hidden",
    }
  }
>
  <span>
    Hello, I'm a tooltip!
  </span>
</div>
`;
